src/main/java/sirius/kernel/xml/SOAPClient.java

```diff
@@ -234,6 +237,18 @@ public SOAPClient withResultTransfomer(BiFunction<StructuredNode, String, Struct
         return this;
     }
 
+    /**
+     * Makes it possible to use a supplier to determine whether the debug logging is active.
+     * The Logger must be on FINE level for this to have any effect.
+     *
+     * @param isDebugLogActive the supplier to determine the active state
+     * @return the client itself for fluent method calls
+     */
+    public SOAPClient withIsDebuglogActive(BooleanSupplier isDebugLogActive) {
```

Comment:
maybe simplify the method name and underlying var to `withDebugMode` or the like?